internal/jstest/jstest.go

```diff
@@ -85,8 +83,7 @@ func Run(ctx context.Context, dir string, args []string) (*internal.TestResults,
 		out  []byte
 	}
 
-	// tokens is a counting semaphore used to enforce a limit of 20 concurrent calls to runShellWithScript.
-	tokens := make(chan struct{}, 20)
+	sema := make(chan struct{}, workers)
```

Comment:
Now that we know the number of workers, I think we should change that code to launch that number of worker goroutines and send work items (files) to them via a channel, not launch one temporary goroutine per work item. That should simplify that code.

Reply:
But how do we ensure we process all the files if the number of workers is less than the number of files?

After we've finished iterating the workers there will still be files left in the channel to be processed, no?

Reply:
I think I know what to do.

Reply:
I think this pattern is more optimal because of long-lived goroutines and the avoidance of excessive concurrency.

Reply:
The sender should close the working channel, workers should exit when the working channel is empty and closed, `sync.WaitGroup` should be used to wait for all of them to exit.

It is a bit more efficient (but only a little bit because goroutines are much more effective than, say, OS threads), but simplicity is the main reason – it is a very common and recognizable pattern in Go.